### PR TITLE
PHP 7.1 cleanups

### PR DESCRIPTION
**Pull request by dbu**
Some leftover from #521 

## Subject

Cleanup tests a bit with :void return type declarations.

### CODE REVIEW

File: docs/reference/document_tree.rst [hunk 1]
```diff
@@ -220,7 +220,7 @@ like this:
     .. code-block:: jinja
 
         {% render(controller(
-            'sonata.admin.doctrine_phpcr.tree_controller:treeAction',
+            'sonata.admin.doctrine_phpcr.tree_controller::tree',
```
Comment (dbu) [inline docs/reference/document_tree.rst]: don't we still need the "Action" suffix with the double colon notation?

Comment (dbu) [reply docs/reference/document_tree.rst]: according to https://symfony.com/doc/current/best_practices/controllers.html#controller-action-naming the `Action` suffix is deprecated. however, the thing here must match the method name, and you are not adjusting the method name in this PR.

changing the method name is a BC break. i suggest we make a ticket here for this, first adding the method and make the `Action` methods call those method and trigger a deprecation warning, then later remove it when we do the next major version. but its a bit out of scope of this PR, and this PR is big enough as is.

**Comment by OskarStark** [comment]
@dbu there are a lots of commits, can you please get rid of them?

Or apply these changes against the PR of @ElectricMaxxx so his PR gets updated, which could then be merged, and we don't los the comments there.

**Comment by dbu** [comment]
yeah, was just thinking the same :-D i will squash that thing

**Comment by ElectricMaxxx** [comment]
@dbu can ammend a more meaningfull commit message pls?

**Comment by ElectricMaxxx** [comment]
Ah, the last commit wasn't the squash result :-)

but at the end we should have one commit  explaining the whole change :-)

**Comment by dbu** [comment]
as #512, this PR goes against the 2.x branch, but it contains commits that are only in master. i guess i throw out those master commits in the rebase?

File: .gitattributes [hunk 2]
```diff
@@ -2,3 +2,4 @@
 *.md export-ignore
 tests/* export-ignore
 docs/* export-ignore
+tests/Fixtures/App/var* export-ignore
```
Comment (OskarStark) [inline .gitattributes]: This one gets overwritten by `dev-kit` on the next run:
https://github.com/sonata-project/dev-kit/blob/master/project/.gitattributes.twig

File: .gitignore [hunk 3]
```diff
@@ -2,7 +2,8 @@
 /docs/_build
 /vendor
 .php_cs.cache
+.phpunit
```
Comment (OskarStark) [inline .gitignore]: do we need this?

File: .travis/check_relevant_lint.sh [hunk 4]
```diff
@@ -1,4 +1,4 @@
-#!/usr/bin/env bash
+#!/usr/bin/env sh
```
Comment (OskarStark) [inline .travis/check_relevant_lint.sh]: This one gets overwritten to `bash` by `dev-kit` on the next run.

Why did you change this?

File: .travis/check_relevant_docs.sh [hunk 5]
```diff
@@ -1,4 +1,4 @@
-#!/usr/bin/env bash
+#!/usr/bin/env sh
```
Comment (OskarStark) [inline .travis/check_relevant_docs.sh]: This one gets overwritten to `bash` by `dev-kit` on the next run.

Why did you change this?

File: .travis/check_relevant_test.sh [hunk 6]
```diff
@@ -1,4 +1,4 @@
-#!/usr/bin/env bash
+#!/usr/bin/env sh
```
Comment (OskarStark) [inline .travis/check_relevant_test.sh]: This one gets overwritten to `bash` by `dev-kit` on the next run.

Why did you change this?

File: .travis/before_script_test.sh [hunk 7]
```diff
@@ -1,3 +1,10 @@
-#!/usr/bin/env bash
+#!/usr/bin/env sh
```
Comment (OskarStark) [inline .travis/before_script_test.sh]: This one gets overwritten to `bash` by `dev-kit` on the next run.

Why did you change this?

File: .travis/before_script_test.sh [hunk 8]
```diff
@@ -1,3 +1,10 @@
-#!/usr/bin/env bash
```
Comment (OskarStark) [inline .travis/before_script_test.sh]: Please revert the CHMOD

File: .travis/before_script_test.sh [hunk 9]
```diff
@@ -1,3 +1,10 @@
...
+#!/usr/bin/env sh
 set -ev
 
+TESTING_SCRIPTS_DIR=vendor/symfony-cmf/testing/bin
+CONSOLE=${TESTING_SCRIPTS_DIR}/console
+
+export KERNEL_CLASS="Sonata\\DoctrinePHPCRAdminBundle\\Tests\\Fixtures\\App\\Kernel"
+echo '+++ create PHPCR +++'
+${CONSOLE} doctrine:phpcr:init:dbal --drop --force -vvv
+${CONSOLE} doctrine:phpcr:repository:init -vvv
```
Comment (OskarStark) [inline .travis/before_script_test.sh]: The whole block gets overwritten by `dev-kit` on the next run.

This should be added [here](https://github.com/sonata-project/dev-kit/blob/master/project/.travis/before_script_test.sh.twig) with a condition on the project.

File: phpunit.xml.dist [hunk 10]
```diff
@@ -11,19 +11,11 @@
          syntaxCheck="false"
```
Comment (OskarStark) [inline phpunit.xml.dist]: as `phpunit.xml.dist` file is managed by `dev-kit`, I am not sure what will survive after the next run

File: docs/reference/document_tree.rst (hunk 1 above)
Comment (dbu) [reply docs/reference/document_tree.rst]: created #537 and changed the doc back to use `Action`

**Comment by dbu** [comment]
there we go. squashed the thing and removed the 3.x commit. please look over it again and merge if you are happy.

**Comment by dbu** [comment]
@ElectricMaxxx would you mind porting the CI changes to the dev-kit?

**Comment by ElectricMaxxx** [comment]
@dbu that is quite a good question, as i heard it isn't possible to have separate travis scripts in sonata-dev-kit, isn't it?

**Comment by OskarStark** [comment]
You can do this, we are doing it here for example:
https://github.com/sonata-project/dev-kit/blob/master/project/.travis/before_install_test.sh.twig

File: .travis/before_script_test.sh (hunk 8 above)
Comment (OskarStark) [reply .travis/before_script_test.sh]: You are right, I thought it was done by mistake.

We can keep this, I added a PR on dev-kit for the other bundle 👍 
https://github.com/sonata-project/dev-kit/pull/440

**Comment by dbu** [comment]
@ElectricMaxxx i cleaned this up again. can you maybe port the things we need to be in the dev-kit there so that we can merge this and won't lose changes on the next dev-kit run?

File: .travis/before_script_test.sh (hunk 9 above)
Comment (ElectricMaxxx) [reply .travis/before_script_test.sh]: thougt dev-kit still got it in?

**Comment by ElectricMaxxx** [comment]
I created: https://github.com/sonata-project/dev-kit/pull/458

File: .gitattributes (hunk 2 above)
Comment (OskarStark) [reply .gitattributes]: please add this to the dev-kit PR, too

File: phpunit.xml.dist (hunk 10 above)
Comment (ElectricMaxxx) [reply phpunit.xml.dist]: no it isn't :-)

Comment (ElectricMaxxx) [reply phpunit.xml.dist]: https://github.com/sonata-project/dev-kit/blob/master/config/projects.yml#L189

Comment (OskarStark) [reply phpunit.xml.dist]: You are right 👍

**Comment by dbu** [comment]
i rebased now that #521 is merged and this is all that is left. mostly re-adding the `: void` to test methods. not sure if its worth it but its correct, and for tests there are no BC concerns.

**Comment by OskarStark** [comment]
Can you please update the PR header and the title accordingly?

**Comment by dbu** [comment]
done. i removed the changelog entry, i think this is not worth putting anything into the changelog ;-)

File: tests/Fixtures/App/DataFixtures/Phpcr/LoadTreeData.php [hunk 11]
```diff
@@ -20,7 +20,7 @@
 
 class LoadTreeData implements FixtureInterface
 {
-    public function load(ObjectManager $manager)
+    public function load(ObjectManager $manager): void
```
Comment (greg0ire) [inline tests/Fixtures/App/DataFixtures/Phpcr/LoadTreeData.php]: Isn't this a BC-break for extending classes?

Comment (OskarStark) [reply tests/Fixtures/App/DataFixtures/Phpcr/LoadTreeData.php]: its a fixture

**Comment by greg0ire** [comment]
Thanks @dbu !
